Tools/cases_generator/analyzer.py

```diff
@@ -1280,6 +1288,70 @@ def analyze_forest(forest: list[parser.AstNode]) -> Analysis:
         instructions, uops, families, pseudos, labels, opmap, first_arg, min_instrumented
     )
 
+#Simple heuristic for size to avoid too much stencil duplication
+def is_large(uop: Uop) -> bool:
+    return len(list(uop.body.tokens())) > 80
+
+def get_uop_cache_depths(uop: Uop) -> Iterator[tuple[int, int, int]]:
+    if uop.name == "_SPILL_OR_RELOAD":
+        for inputs in range(MAX_CACHED_REGISTER+1):
+            for outputs in range(MAX_CACHED_REGISTER+1):
+                if inputs != outputs:
+                    yield inputs, outputs, inputs
+        return
+    if uop.name in ("_DEOPT", "_HANDLE_PENDING_AND_DEOPT", "_EXIT_TRACE"):
+        for i in range(MAX_CACHED_REGISTER+1):
+            yield i, 0, 0
+        return
+    if uop.name in ("_START_EXECUTOR", "_JUMP_TO_TOP", "_COLD_EXIT"):
+        yield 0, 0, 0
+        return
+    if uop.name == "_ERROR_POP_N":
+        yield 0, 0, 0
+        return
+    non_decref_escape = False
+    for call in uop.properties.escaping_calls.values():
+        if "DECREF" in call.call.text or "CLOSE" in call.call.text:
```

Comment:
This is too imprecise. It should exclude `PyStackRef_CLOSE_SPECIALIZED`. Those are unecessary spills.